src/features/users.ts

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable no-param-reassign */
+
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { getUsers } from '../api/users';
+
+export const init = createAsyncThunk('users/fetch', () => getUsers());
+
+type Users = {
+  users: User[];
+  loaded: boolean;
+  hasError: boolean;
+};
+const initialState: Users = {
+  users: [],
+  loaded: false,
+  hasError: false,
+};
+
+const usersSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {},
+  extraReducers(builder) {
+    builder.addCase(init.pending, state => {
+      state.hasError = false;
+      state.loaded = false;
+    });
+
+    builder.addCase(init.fulfilled, (state, action) => {
+      state.users = action.payload;
+      state.loaded = true;
+    });
+
+    builder.addCase(init.rejected, state => {
+      state.hasError = true;
+      state.loaded = true;
+    });
```

Comment:
Consider logging the error or providing more detailed error information in the state when the `init` action is rejected. This can help with debugging and provide more context for error handling in the UI.